setup.py

```diff
@@ -87,142 +64,106 @@ def cmake_version_check():
-
-    def run(self):
-        _ctx.cxx_install_root = self.cxx_install_root
-        _ctx.install_usersite = self.install_usersite
-        install.run(self)
-
-
 class CMakeExtension(Extension):
     def __init__(self, name):
         Extension.__init__(self, name, sources=[])
 
 
 class OTIO_build_ext(setuptools.command.build_ext.build_ext):
+    """
     def initialize_options(self):
-        self.cxx_coverage = False
-        setuptools.command.build_ext.build_ext.initialize_options(self)
+        super(setuptools.command.build_ext.build_ext, self).initialize_options()
+    """
 
     def run(self):
-        _ctx.cxx_coverage = (
-            self.cxx_coverage is not False
-            or bool(os.environ.get("OTIO_CXX_COVERAGE_BUILD"))
-        )
-
+        # This works around the fact that we build _opentime and _otio
+        # extensions as a one-shot cmake invocation. Usually we'd build each
+        # separate using build_extension.
         self.build()
 
     def build(self):
-        _ctx.ext_dir = os.path.abspath(self.build_lib)
-
-        _ctx.build_temp_dir = (
+        self.build_temp_dir = (
             os.environ.get("OTIO_CXX_BUILD_TMP_DIR")
             or os.path.abspath(self.build_temp)
         )
-        _ctx.debug = self.debug or bool(os.environ.get("OTIO_CXX_DEBUG_BUILD"))
 
-        if not _ctx.ext_dir.endswith(os.path.sep):
-            _ctx.ext_dir += os.path.sep
-        if not os.path.exists(_ctx.build_temp_dir):
-            os.makedirs(_ctx.build_temp_dir)
+        if not os.path.exists(self.build_temp_dir):
+            os.makedirs(self.build_temp_dir)
+
+        debug = (self.debug or bool(os.environ.get("OTIO_CXX_DEBUG_BUILD")))
+        self.build_config = ('Debug' if debug else 'Release')
+
+        self.cmake_generate()
+        self.cmake_install()
+
+    def cmake_generate(self):
+        # Determine the target directory for the built library
+        install_dir = os.path.abspath(self.build_lib)
+        if not install_dir.endswith(os.path.sep):
+            install_dir += os.path.sep
+
+        cmake_args = [
+            '-DPYTHON_EXECUTABLE=' + sys.executable,
+            '-DOTIO_PYTHON_INSTALL:BOOL=ON',
+            '-DOTIO_CXX_INSTALL:BOOL=ON',
+            '-DCMAKE_BUILD_TYPE=' + self.build_config,
+            '-DPYBIND11_FINDPYTHON=ON',  # Smart tool to find python's libs
+        ]  # https://pybind11.readthedocs.io/en/latest/compiling.html#findpython-mode
+
+        # install the C++ into the opentimelineio/cxx-sdk directory under the
+        # python installation
+        cmake_install_prefix = os.path.join(
+            install_dir,
+            "opentimelineio",
+            "cxx-sdk"
+        )
+
+        cmake_args += [
+            '-DOTIO_PYTHON_INSTALL_DIR=' + install_dir,
+            '-DCMAKE_INSTALL_PREFIX=' + cmake_install_prefix,
+        ]
+
+        if platform.system() == "Windows":
+            if sys.maxsize > 2**32:
+                cmake_args += ['-A', 'x64']
```

Comment:
Migrated from @meshula on [previous PR](https://github.com/PixarAnimationStudios/OpenTimelineIO/pull/955#discussion_r621901333):
> per usd_build.py, we should detect VS2019 or greater, as this argument doesn't apply for VS2017 and less, where 64 bit builds are indicated by appending "Win64" to the build target.

Reply:
Alternatively we could add a note to our readme that currently vs2019 or later is required.

Reply:
VFX Platform seems to support minimum vs2017 across the board, FWIW

Reply:
In this re-factor I didn't actually change any of the build behavior beyond re-targeting the locations build products land in.

On this issue, I was thinking we address this when we do the cibuildwheel change because it will exercise a lot of wacky environment combos that will test what we want to address. How does that sound?

Reply:
That's what I was implying by "currently vs2019". It's not a heavy lift to add vs2017 support, but at the same time, it is also not a regression due to this change; so a note in the Readme, now that we've noted the issue, suffices IMO